src/components/gamespec/Element.tsx

```diff
@@ -57,6 +60,9 @@ class Element extends React.Component<ElementProps, ElementState> {
     let index = data.indexOf(key);
     data.splice(index, 1);
     this.setState({ selected_cards: data });
+    if (!this.props.setSelectedElements) {
```

Comment:
Maybe check for both in the start itself. There is no point in reaching this step is `setSelectedElements` is undefined.